deps/captnlog/src/captnlog.c

```diff
@@ -123,6 +120,9 @@ void captainslog_log(int level, const char *file, int line, const char *fmt, ...
         file_str[0] = '\0';
     }
 
+    /* Lock the mutex to prevent multiple threads trying to write at the same time */
+    lock();
```

Comment:
I moved this down because there is no need to lock already when writing to the stack buffers.